ui/src/App.js

```diff
@@ -133,22 +136,45 @@ class App extends Component {
     datasetApi.datasetGet(datasetCallback);
   }
 
+  getFacetMap(facets) {
+    var facetMap = new Map();
+    facets.forEach(function(facet) {
+      facetMap.set(facet.es_field_name, facet);
+    });
+    return facetMap;
+  }
+
   handleSearch(searchResults) {
+    // If a facet is selected from the dropdown, searchResults is the selected facet.
+    // If a facetValue is selected from the dropdown, searchResults is an Array
+    // of all selected facet values.
     let extraFacetEsFieldNames = this.state.extraFacetEsFieldNames;
+    // An array of selected facet values, of the form esFieldName=facetValue.
```

Comment:
A lot of this code is obsolete, from the field search days.

Instead of looping through searchResults, you only need to look at the last element. That's the drop-down row that was just selected.

Reply:
handleChange is also triggered when a chip is deleted, so looking at the last element alone will not work.

Reply:
react-select [provides action type](https://react-select.com/advanced#action-meta), which makes the code even easier to understand.
```
handleSearchBoxChange(selectedOptions, action) {
  if action == "clear"
    // x on right of search box was clicked.
    clearFacetValueSelections()
    Or just write code inline
  else if action == "remove-value"
    // chip x was clicked.
    // selectedOptions contains remaining selected facet values (ie remaining chips).
    Figure out which chip was deleted and call updateFacets()
  else if action == "select-option"
    // Drop-down row was clicked.
    // selectedOptions contains current chips, plus one additional element representing the drop-down row that was clicked.
    Only look at last element of selectedOptions.
    Add last element's esFieldName to this.state.extraFacetEsFieldNames.
    Call this.facetsApi.facetsGet()
}
```